hazelcast/src/main/java/com/hazelcast/jet/impl/JetClientInstanceImpl.java

```diff
@@ -126,6 +143,83 @@ public Job newJobProxy(long jobId, boolean isLightJob, @Nonnull Object jobDefini
         return new ClientJobProxy(client, jobId, isLightJob, jobDefinition, config);
     }
 
+    @Override
+    public void submitJobFromJar(@Nonnull SubmitJobParameters submitJobParameters) {
+
+        try {
+            UUID sessionId = UuidUtil.newSecureUUID();
+
+            Path jarPath = submitJobParameters.getJarPath();
+            // Calculate some parameters for JobMetadata
+            String fileNameWithoutExtension = getFileNameWithoutExtension(jarPath);
+            String sha256Hex = calculateSha256Hex(jarPath);
+            long jarSize = Files.size(jarPath);
+
+            // Send job meta data
+            boolean result = sendJobMetaData(sessionId, fileNameWithoutExtension, sha256Hex, submitJobParameters);
+            if (result) {
+                logFine(getLogger(), "Submitted JobMetaData successfully for jarPath: %s", jarPath);
+            }
+            // Send job parts
+            sendJobMultipart(jarPath, sessionId, jarSize);
+
+        } catch (IOException | NoSuchAlgorithmException exception) {
+            sneakyThrow(exception);
+        }
+    }
+
+    protected String getFileNameWithoutExtension(Path jarPath) {
+        String fileName = jarPath.getFileName().toString();
+        if (!fileName.endsWith(".jar")) {
+            throw new JetException("File name extension should be .jar");
+        }
+        fileName = fileName.substring(0, fileName.lastIndexOf('.'));
+        return fileName;
+    }
+
+    public String calculateSha256Hex(Path jarPath) throws IOException, NoSuchAlgorithmException {
```

Comment:
IMHO unnecessary; better to rename method in utils, as suggested above. You can also use static import for that method

Reply:
Ok